### ISSUE TITLE
Add `quarkus.timezone` to make time zone selection deterministic

### ISSUE THREAD

**Issue by kdubb** [quarkusio/quarkus#43578]
### Description

You can read about my catalyzing issue at https://github.com/quarkusio/quarkus/issues/43577.
Additionally, numerous similar issues have been filed over the years. 

Also, when you have figured out the issue with initialization and still want to set the time zone, you need to track down where that should happen; spoiler... it's at least two or three places to make sure the time zone is set during Dev, Test, Continuous Testing**, and Prod.

It's worth noting that Continuous Testing and `quarkus test` were where the real problems arose; I believe this is due to forking for tests, but I'm not sure.

Currently, this simple task can produce hours of spelunking and debugging to get done. It feels like a nightmare.

Finally, our issues did arise from a bug, but it's worth noting that there's no workaround. We abandoned using `@CreationTimestamp` and `@UpdateTimestamp` altogether because we couldn't get the clock set in certain modes.

### Implementation ideas

Add a `quarkus.timezone` property that, when set, directs Quarkus to set the Clock prior to STATIC_INIT.

This would allow developers to have a simple property to set and know that in all configurations, the time zone will be what they require, is appropriately set, and is at the right time. 

Additionally, given that Quarkus has a very dynamic configuration system... setting `%test.quarkus.timezone=UTC` would give you a simple method to enable it only for tests and continuous testing.

**Comment by codespearhead** [comment]
If the suggested solution at https://github.com/quarkusio/quarkus/issues/43577 works, you just need to set `quarkus.hibernate-orm.jdbc.timezone` accordingly for each environment.

**Comment by geoand** [comment]
@kdubb I completely agree that the current experience is very subpar, and I am assuming we could tackle it without too much trouble. 
It would be great if you could provide some kind of sample that tests the various modes where you've encountered the problem - that way we would have a proper test to use for coming with a solution.

**Comment by chainhead** [comment]
When running as a container, I set an environment variable `TZ` as say, `Asia/Kolkata` and that seems to solve the issue.

**Comment by cyChop** [comment]
Just sharing my current tests: I was loading the ZoneId from a config properties, but it implied injecting it everywhere and not forget. So I wrote this test:

```java
System.out.println(ZoneId.systemDefault()); // => Europe/Paris
TimeZone.setDefault(TimeZone.getTimeZone("Europe/Berlin"));
System.out.println(ZoneId.systemDefault()); // => Europe/Berlin
```

So now I can use ZoneId.systemDefault with confidence. It may not cover all of the OP's original concerns, but it may provide some grounds for implementation, even when the `TZ` var is not an option.
